CustomScript/customscript.py

```diff
@@ -368,7 +374,13 @@ def download_external_file(uri, command, hutil):
     file_path = os.path.join(download_dir, file_name)
     try:
         download_and_save_file(uri, file_path)
-        preprocess_files(file_path, hutil)
+        enable_dos_converting = True
+        public_settings = hutil.get_public_settings()
+        if public_settings:
```

Comment:
lines 378-383 are repeated exactly the same in both methods, please extract them into a single place and pass it as a flag to these methods.